src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java

```diff
@@ -186,6 +279,57 @@
  *   &lt;property name=&quot;setterCanReturnItsClass&quot; value=&quot;true&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <pre>
+ * public class SomeClass {
+ *
+ *   private String field; // field 1
+ *   private String test; // field 2
+ *
+ *   // Parameterized constructor
+ *   public SomeClass(String test) { // violation, 'test' param hides 'test' field
+ *   }
+ *   public void method(String param) { // OK
+ *       String field = param; // violation, 'field' variable hides 'field' field
+ *   }
+ *   // setter method
+ *   public void setTest(String test) { // OK, 'test' param doesn't hide any field
+ *       this.field = field;
+ *   }
+ *   // chain-setters method
+ *   public SomeClass setField(String field) { // OK, 'field' param doesn't hide any field
+ *       this.field = field;
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check so that it ignores parameters of abstract methods:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;HiddenField&quot;&gt;
+ *   &lt;property name=&quot;ignoreAbstractMethods&quot; value=&quot;true&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * // abstract class
+ * abstract class SomeClass {
+ *
+ *   private String field; // field
+ *
+ *   // Parameterized Constructor
+ *   public SomeClass(int field) { // violation, 'field' param hides a 'field' field
+ *     float field; // violation, 'field' variable hides a 'field' field
+ *   }
+ *   // abstract method
+ *   public abstract int method(String field); // OK
+ * }
+ * // Regular class extends abstract class
+ * public class Demo extends SomeClass {
```

Comment:
fix indentation

Reply:
what's wrong ?

Reply:
ups, sorry, it is next class, it is ok in this case.